xonsh/xontribs.py

```diff
@@ -109,18 +109,25 @@ def xontrib_metadata():
 def xontribs_load(names, verbose=False):
     """Load xontribs from a list of names"""
     ctx = builtins.__xonsh__.ctx
+    res = 0
     for name in names:
         if verbose:
             print("loading xontrib {0!r}".format(name))
-        update_context(name, ctx=ctx)
+        try:
+            update_context(name, ctx=ctx)
+        except Exception:
+            res = 2
+            print_exception("Failed to load xontrib {}.".format(name))
     if update_context.bad_imports:
```

Comment:
I know not your code, but should this be `hasattr(update_context, 'bad_imports')` or something similar?

Reply:
Yes, it should. I didn't notice.

Reply:
Done.